src/boot/efi/boot.c

```diff
@@ -2222,6 +2270,125 @@ static void config_entry_add_linux(
         }
 }
 
+static EFI_STATUS secure_boot_enroll(EFI_FILE *root_dir, Config *config, ConfigEntry *entry) {
+        assert(root_dir);
+        assert(config);
+        assert(entry);
+
+        typedef struct {
+                const CHAR16 *name;
+                const CHAR16 *filename;
+                const EFI_GUID vendor;
+                CHAR8 *buffer;
+                UINTN size;
+        } SecureBootVariable;
+
+        static SecureBootVariable sb_vars[] = {
+                { L"db",  L"db.esl",  EFI_IMAGE_SECURITY_DATABASE_VARIABLE, NULL, 0 },
+                { L"KEK", L"KEK.esl", EFI_GLOBAL_VARIABLE, NULL, 0 },
+                { L"PK",  L"PK.esl",  EFI_GLOBAL_VARIABLE, NULL, 0 },
```

Comment:
Are we sure the `.esl` files is what goes into these vars? When I enroll keys into ovmf manually, it only accepts the crt files (though, admittedly, different enroll process).

Also, is the order important? (I really should go read the spec on this :joy_cat:).

Reply:
Yes.

When you enroll through the firmware UI, the firmware actually does some processing. The actual format of what you need to submit is detailed in the UEFI spec (https://uefi.org/sites/default/files/resources/UEFI_Spec_2_8_final.pdf) page 1714, _32.4 Firmware/OS Key Exchange: passing public keys_.